smartapps/cuchen-wifi-cooker/cuchen-wifi-cooker.src/cuchen-wifi-cooker.groovy

```diff
@@ -0,0 +1,242 @@
+  }
+}
+
+def warm(childDevice) {
+  log.debug "(app) Executing 'warm'"
+  if (!atomicState?.lastStatus?.trim()) {
+      atomicState.lastStatus = childDevice.device.currentValue("status")
+      atomicState.retry = 0
+      atomicState.nextStatus = "warm"
+      sendCommand(settings.cookerId, "setWarm", [passwd: settings.password, type:0, isPlus:true, tempr:1])
+  }
+}
+
+def reheat(childDevice) {
+  log.debug "(app) Executing 'reheat'"
+  if (!atomicState?.lastStatus?.trim()) {
+      atomicState.lastStatus = childDevice.device.currentValue("status")
+      atomicState.retry = 0
+      atomicState.nextStatus = "reheat"
+      sendCommand(settings.cookerId, "reheat", [passwd: settings.password])
+  }
+}
+
+def autoClean(childDevice) {
+  log.debug "(app) Executing 'autoClean'"
+  if (!atomicState?.lastStatus?.trim()) {
+      atomicState.lastStatus = childDevice.device.currentValue("status")
+      atomicState.retry = 0
+      atomicState.nextStatus = "auto cleaning"
+
+      //def now = Calendar.instance
+      //sendCommand(settings.cookerId, "setTime", [passwd: settings.password, mon: now[MONTH], day: now[DATE], hour: now[HOUR], min: now[MINUTE], isAm: (now[AM_PM] == AM)])
+      sendCommand(settings.cookerId, "autoClean", [passwd: settings.password])
+  }
+}
+
+def cancel(childDevice) {
+  log.debug "(app) Executing 'cancel'"
+  if (!atomicState?.lastStatus?.trim()) {
+      atomicState.lastStatus = childDevice.device.currentValue("status")
+      atomicState.retry = 0
+      atomicState.nextStatus = "stand by"
+      sendCommand(settings.cookerId, "cancel", [passwd: settings.password])
+  }
+}
+
+// function for axon
+def retryMax() {
+	return 5;
+}
+
+def getRequest(path, deviceId, apiName, params) {
+  def axonServer = "http://axon-iot.com"
+  def axonSvcKey = "1422266982285"
+  def axonApiKey = "cbce8f97dd26fca681151996882672b129351ce6923259e022b785beb7d7de"
+  def axonClientId = "947D9B3E-2CB5-4B26-99D9-F300F7171A05"
+
+  def paramJson = JsonOutput.toJson(params)
+  if(paramJson == "\"\"") {
+    paramJson = ""
+  }
+  def body = "apiKey=${axonApiKey}&svcKey=${axonSvcKey}&clientId=${axonClientId}&deviceId=${deviceId}&apiName=${apiName}&params=${paramJson}"
+
+  def request = [
+    uri : axonServer,
+    path : path,
+    contentType : "application/json",
+    body : body
+  ]
+
+  return request
+}
+
+def getStatus(deviceId, once) {
+  def path = "/api/getStatus.action"
+  def request = getRequest(path, deviceId, "", "")
+  //log.debug "request : $request"
+  
+  def dev = getChildDevice(deviceId)
+  try {
+      httpPost(request) { resp ->
+        def result = resp.data.success
+        log.debug "response data: ${resp.data}"
+        
+        dev?.updateStatus(resp.data.bean)
+        def temp = resp.data.bean[3]
+        def status = dev?.device.currentValue("status")
+        if (!once && atomicState.retry < retryMax() && status == atomicState.lastStatus) {
```

Comment:
Why is this based on the status of the device rather than the response? The way this is implemented will have it send up to 5 more POST requests to the server even if the server is returning 200. This conditional statement is checking whether or not the state has been sent to the device, so I would separate this out from the HTTP request and maybe handle this as updateDevice(status, x, y, ...). You should only be making additional calls out if the response isn't returning 200.